libraries/ESP8266WiFi/src/WiFiClient.cpp

```diff
@@ -264,19 +311,20 @@ size_t WiFiClient::peekBytes(uint8_t *buffer, size_t length) {
     return _client->peekBytes((char *)buffer, count);
 }
 
-void WiFiClient::flush()
+bool WiFiClient::flush(unsigned int maxWaitMs)
 {
-    if (_client)
-        _client->wait_until_sent();
+    return !_client || _client->wait_until_sent(maxWaitMs?: WIFICLIENT_MAX_FLUSH_WAIT_MS);
```

Comment:
I get the shorthand, but using the early-termination feature to avoid an if-then could be confusing